src/pages/guides/using/creating_rest_apis.md

```diff
@@ -153,6 +153,145 @@ curl -i -X OPTIONS https://adobeioruntime.net/...
+            "read:pets": "read your pets"
+          }
+        }
+    }
+}
+```
+This enables scope validation for the API endpoint, allowing requests with access tokens that have the scopes `write:pets` OR `read:pets`.
+>Ensure that there is a `securityDefinitions` defined with type `oauth2`. You can use any authorization provider by providing a `authorizationUrl`, however if left empty, IMS, will be used as default.
+
+After publishing the Swagger file, you can use this endpoint to call the action `your-namespaces/default/my-require-validation-web-action` as follows: 
+```bash
+curl -i -H "Authorization: Bearer <ims_access_token>" https://guest.adobeioruntime.net/api/v2/ims-validation-endpoint
+```
+
+You can also enable `client_id` validation by adding `clientIds` to the security object, like this:
+```json
+{
+    "basePath": "/v2",
+    "paths": {
+      "/ims-validation-endpoint": {
+        "get": {
+          "operationId": "your-namespaces/default/my-require-validation-web-action.json",
+          "security": [
+            {
+              "scopes_auth": {
+                "clientIds": [
+                  "zookeeper",
+                  "dogwalker"
+                ]
+              }
+            }
+          ]
+        }
+      }
+    },
+    "securityDefinitions": {
+        "scopes_auth": {
+          "type": "oauth2",
+          "authorizationUrl": "https://ims-na1-stg1.adobelogin.com/ims/validate_token/v1?client_id=ioruntime&type=access_token",
+          "flow": "implicit",
+          "scopes": {
+            "write:pets": "modify pets in your account",
+            "read:pets": "read your pets"
+          },
+          "clientIds": {
+            "zookeeper": "Client ID for Zookeeper",
+            "dogwalker": "Client ID for Dogwalker"
+          }          
+        }
+    }
+}
+```
+This configuration allows the action to accept requests with access tokens that have the client IDs `zookeeper` OR `dogwalker`.
+
+> Note that you can use both `scopes` and `clientIds` simultaneously. In this case, the action will accept requests with access tokens that have both the scopes `write:pets` OR `read:pets` AND the client IDs `zookeeper` OR `dogwalker`.
+
+### Basic Authentication
+
 You secure an API the same way you&rsquo;d do it for web actions. You can read more about this on the [Securing Web Actions](securing_web_actions.md) page.
 
-Once you&rsquo;ve enabled basic authentication for an action, you&rsquo;d have to pass the `X-Require-Whisk-Auth` header and the secret you chose when making an API call. 
+Once you&rsquo;ve enabled basic authentication for an action, you&rsquo;d have to pass the `X-Require-Whisk-Auth` header and the secret you chose when making an API call.
+
+### IP Whitelisting / Blacklisting
+
+Endpoints can also be configured to only allow/block requests from specific IP addresses. This can be done by adding the `ip_whitelist` or `ip_blacklisting` to the `security` definition as follows:
```

Comment:
Why `ip_whitelist` (no `-ing` ending) and `ip_blacklisting` (with `-ing` ending)?

Reply:
Thx for noticing, it's a typo. It should `ip_whitelist`. Changing.